ext/drx/drx.h

```diff
@@ -435,6 +435,19 @@ DR_EXPORT
 size_t
 drx_buf_get_buffer_size(void *drcontext, drx_buf_t *buf);
 
+
+DR_EXPORT
+/**
+ * Pads a basic block with a label for routines which rely on inserting instrumentation
+ * after every instruction. Note that users of this routine must act on the previous
+ * instruction before the is-app check afterwards, because the label is not marked as an
+ * app instruction.
+ *
+ * \returns whether terminator was introduced.
```

Comment:
Still working on this, but I'm wondering now if it's really worth always inserting instrumentation at the end of the basic block. In the case of an exit cti, even if a user *wants* to insert instrumentation after a call or a syscall, core DR won't let them because the exit cti must terminate the basic block. Also there's the problem with drreg -- it will lazily restore registers at the end of the basic block, meaning just before the label instruction. Even drreg's instrumentation causes core DR to fail because the restoration happens after a cti, on which my progress to fully fix has been slow-going.

Should we revert to only inserting instrumentation if the basic block was already truncated?